docs/infra/web-application-firewall.md

```diff
@@ -0,0 +1,57 @@
+# Web Application Firewall (WAF)
+
+The infrastructure template includes support for AWS Web Application Firewall (WAF) to protect application load balancers from common web threats. This document describes how WAF is configured. The WAF setup process will:
+
+1. Create an AWS WAF web ACL with managed rule sets in the network layer
+2. Configure WAF logging to CloudWatch
+3. Associate the WAF with application load balancers in the service layer
+
+## How WAF works
+
+WAF helps protect your web applications from common web exploits and bots that could affect availability, compromise security, or consume excessive resources. The template-infra implementation uses AWS managed rule sets that provide protection against common threats:
+
+- **AWS Common Rule Set** - Provides general protection against common web threats like SQL injection, cross-site scripting (XSS), and other OWASP Top 10 vulnerabilities
+- **AWS Known Bad Inputs Rule Set** - Blocks request patterns that are known to be invalid and are associated with exploitation or discovery of vulnerabilities
+
+The WAF is created at the network level and can be associated with application load balancers in the service layer.
+
+## Requirements
+
+There are no specific prerequisites for enabling WAF.
+
+## 1. Enable WAF in application config
+
+WAF is enabled by default in the template. You can find and modify this setting in your application's `app-config` module (`infra/<APP_NAME>/app-config/main.tf`):
+
+```terraform
+enable_waf = true
+```
+
+Set `enable_waf = false` if you want to disable WAF protection.
+
+## 2. Deploy the WAF in the network layer
+
+The WAF must be created in the network layer before it can be associated with load balancers in the service layer. Note that the WAF may already have been deployed as part of the [network setup process](./set-up-network.md). If not, run the following command to deploy the WAF:
+
+```bash
+make infra-update-network NETWORK_NAME=<NETWORK_NAME>
+```
+
+## 3. Associate WAF with application load balancers
+
+After the WAF is created in the network layer, run the following command to associate it with your application's load balancer:
+
+```bash
+make infra-update-app-service APP_NAME=<APP_NAME> ENVIRONMENT=<ENVIRONMENT>
+```
+
+## 4. View WAF logs
```

Comment:
Add a verification step before this step to test the WAF.

You can use one or both of the following curl calls which should trigger the WAF's XSS protection

```
service_endpoint="$(terraform -chdir="infra/${app_name}/service" output -raw service_endpoint)"
curl -k "${service_endpoint}?search=<script>alert(1)</script>"
```
or
```
curl -k -X POST "${service_endpoint}" \
  -H "Content-Type: application/json" \
  -d '{"bad_input": "<script>alert(1)</script>"}'
```

both should return 403 forbidden